dali/operators/generic/one_hot.cc

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2017-2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/generic/one_hot.h"
+
+namespace dali {
+
+void OneHot::RunImpl(HostWorkspace &ws) {
+  auto &output = ws.OutputRef<CPUBackend>(0);
+  auto &input = ws.InputRef<CPUBackend>(0);
+  for (int i = 0; i < batch_size_; ++i) {
+    auto &inptr = input[i];
+    auto cls = inptr.template mutable_data<int>();
+    DALI_ENFORCE(*cls < nclasses_, "Values are bigger than specified number of classes");
+    auto &outptr = output[i];
+    auto one_hot = outptr.template mutable_data<int>();
+    one_hot[*cls] = 1;
+  }
+}
+
+DALI_REGISTER_OPERATOR(OneHot, OneHot, CPU);
+
+DALI_SCHEMA(OneHot)
+    .DocStr(
```

Comment:
I would check this API https://www.tensorflow.org/api_docs/python/tf/one_hot (on and off values, axis as well).